src/ui/Windows/ThisMfcApp.cpp

```diff
@@ -1474,39 +1471,42 @@ void ThisMfcApp::GetLanguageFiles()
     // Create LCID
     LCID lcid = MAKELCID(FoundResLangID, SORT_DEFAULT);
 
-    inum = ::GetLocaleInfo(lcid, LOCALE_SNATIVELANGNAME, NULL, 0);
+    inum = ::GetLocaleInfo(lcid, LOCALE_SNATIVELANGNAME, nullptr, 0);
     if (inum > 0) {
+      std::unique_ptr<wchar_t[]> szLanguage_English;
+      std::unique_ptr<wchar_t[]> szCountry_Native;
+      std::unique_ptr<wchar_t[]> szCountry_English;
+
       // Get language name in that language
-      szLanguage_Native = new wchar_t[inum + 1];
-      ::GetLocaleInfo(lcid, LOCALE_SNATIVELANGNAME, szLanguage_Native, inum);
+      const auto szLanguage_Native = std::make_unique<wchar_t[]>(inum + 1);
+      ::GetLocaleInfo(lcid, LOCALE_SNATIVELANGNAME, szLanguage_Native.get(), inum);
 
       int jnum = 0;
       if (!cs_CC.IsEmpty()) {
         // Get Country name in that language
-        jnum = ::GetLocaleInfo(lcid, LOCALE_SNATIVECTRYNAME, NULL, 0);
+        jnum = ::GetLocaleInfo(lcid, LOCALE_SNATIVECTRYNAME, nullptr, 0);
         if (jnum > 0) {
-          szCountry_Native = new wchar_t[jnum + 1];
-          ::GetLocaleInfo(lcid, LOCALE_SNATIVECTRYNAME, szCountry_Native, jnum);
+          szCountry_Native = std::make_unique<wchar_t[]>(jnum + 1);
+          ::GetLocaleInfo(lcid, LOCALE_SNATIVECTRYNAME, szCountry_Native.get(), jnum);
         }
         // Get Country name in English
         int knum = ::GetLocaleInfo(lcid, LOCALE_SENGCOUNTRY, NULL, 0);
         if (knum > 0) {
-          szCountry_English = new wchar_t[knum + 1];
-          ::GetLocaleInfo(lcid, LOCALE_SENGCOUNTRY, szCountry_English, knum);
+          szCountry_English = std::make_unique<wchar_t[]>(knum + 1);
+          ::GetLocaleInfo(lcid, LOCALE_SENGCOUNTRY, szCountry_English.get(), knum);
         }
       }
 
       // Now try to make first character of language name in that language upper case
       // Should use LCMapStringEx but that is for Vista + and we support XP & later
-      wchar_t *szLanguage_NativeUpper(NULL);
       int iu = LCMapString(lcid, LCMAP_LINGUISTIC_CASING | LCMAP_UPPERCASE,
-                           szLanguage_Native, inum,
-                           szLanguage_NativeUpper, 0);
+                           szLanguage_Native.get(), inum,
```

Comment:
This seems like a change in behavior - is ‘szLanguage_Native’ guaranteed to be null here?

Reply:
[LCMapString()](https://learn.microsoft.com/en-us/windows/win32/api/winnls/nf-winnls-lcmapstringw) ignores the `lpDestStr` argument (the next-to-last one) if the `cchDest` (the last argument) is zero:
> The application can set cchDest to 0. In this case, the function does not use the lpDestStr parameter and returns the required buffer size for the mapped string or sort key.